cmd/istioctl/main.go

```diff
@@ -223,3 +247,64 @@ func readInput() (proto.Message, error) {
 
 	return v, nil
 }
+
+// readInput reads multiple documents from the input and checks with the schema
+func readInputs() ([]MetadataAndRule, error) {
+
+	var reader io.Reader
+	var err error
+
+	if file == "" {
+		reader = os.Stdin
+	} else {
+		reader, err = os.Open(file)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	var varr []MetadataAndRule
+
+	// We store route-rules as a YaML stream; there may be more than one decoder.
+	var yamlDecoder *yaml.YAMLOrJSONDecoder = yaml.NewYAMLOrJSONDecoder(reader, 512*1024)
+	for {
+		v := MetadataAndRule{}
+		err = yamlDecoder.Decode(&v)
+		if err == io.EOF {
+			break
+		}
+		if err != nil {
+			fmt.Printf("cannot parse proto message: %v", err)
```

Comment:
It would be better to return the error and let the caller handle error handling, e.g. 

```
return nil, fmt.Errorf("cannot parse proto message: %v", err)
```

Similar comment applies to the other instances of os.Exit() below.